### PR TITLE
refactor(header): change source/value of username header

### PR DESCRIPTION
**Pull request by jpetto**
## Goal

update source/value of `username` header sent down to subgraphs.

- add a couple unit tests

## References

JIRA ticket:
* https://getpocket.atlassian.net/browse/BACK-1340

### CODE REVIEW

File: src/jwtUtils.ts [hunk 1]
```diff
@@ -24,6 +31,7 @@ export interface CognitoUser {
   exp: number;
   iat: number;
   email_verified: string;
+  identities: CoginitoUserIdentity[];
```
Comment (Herraj) [inline src/jwtUtils.ts:34]: Wondering why `identities` is an array? Is this just how we get the `identities` from cognito? --> `[{userId: 'Id-A'}, {someOtherId: 'id-B'}, ...]`

**Review by Herraj — APPROVED** [review]
🚀

File: src/jwtUtils.ts (hunk 1 above)
Comment (jpetto) [reply src/jwtUtils.ts:34]: good question - i decoded my JWT and found identities was an array. i'm not sure about the reason or if there's a spec this is adherent to. maybe @bassrock knows?

Comment (bassrock) [reply src/jwtUtils.ts:34]: Hmm. Thats interesting i thought it was a single one.

My guess would be that Cognito is a UserPool, there are multiple ways that a user can be created in a UserPool. Those are our "Providers". My guess is this is the provider user id (which we want) and will have many if the user signed in, in another manner.

However, we can guarantee that our UserPool will only have 1 provider (mozilla sso) and should not be an issue.

**Pull request merged**